api/src/main/java/org/openmrs/module/fhir2/api/search/SearchQuery.java

```diff
@@ -37,10 +39,20 @@
 	 *
 	 * @param theParams search params.
 	 * @param dao generic dao
-	 * @param translator generic translator
+	 * @param translator generic translator In case of $everything operation, package the results in
+	 *            SimpleBundleProvider to include count of _include and _revinclude resources in the
+	 *            total resources count and prevent paging
 	 * @return IBundleProvider
 	 */
 	public IBundleProvider getQueryResults(SearchParameterMap theParams, O dao, V translator, W searchQueryInclude) {
+		if (!theParams.getParameters(FhirConstants.PATIENT_EVERYTHING_SEARCH_HANDLER).isEmpty()) {
+			SimpleBundleProvider result = new SimpleBundleProvider(
+			        new SearchQueryBundleProvider<>(theParams, dao, translator, globalPropertyService, searchQueryInclude)
+			                .getAllResources());
+			
+			result.setPreferredPageSize(result.size());
+			return result;
+		}
```

Comment:
@varung-31 I was thinking along the lines that when in future we implement $everything operation on other resources(eg: Encounter), we'd have to prevent pagination there also, so maybe adding it here would make it useable for $everything on other resources as well.

Reply:
This would make more sense if we generalise the FhirConstants.PATIENT_EVERYTHING_SEARCH_HANDLER to just EVERYTHING_SEARCH_HANDLER and change its value as well

Reply:
I'm not sure why we're trying to wrap things in a `SimpleBundleProvider`... that seems unnecessary when we could just add this logic to the `SearchQueryBundleProvider`'s `preferredPageSize()` implementation? I tend to agree that having a generic `EVERYTHING_SEARCH_HANDLER` seems like the way forward. Alternatively, I suppose this could just be done in the `BaseFhirService`....

Reply:
Yup, we can incorporate the change in the preferred page size method as well by checking for a generic `EVERYTHING_SEARCH_HANDLER `

Reply:
I see. Yes, you're absolutely right. Thanks for the explanation!